### ISSUE TITLE
Should -Throw wildcard charachters

### ISSUE THREAD

**Issue by DarkLite1** [pester/Pester#1003]
## 1. Provide a general summary of the issue in the Title above

It would be nice if the following could be used:
```powershell
{throw 'this is a test'} | Should -Throw "*is*test"
```
## 2. Describe Your Environment

Operating System, Pester version and PowerShell version:

```powershell
Pester version     :  
PowerShell version : 5.1.14409.1012
OS version         : Microsoft Windows NT 6.2.9200.0
```

## 3. Expected Behavior

```powershell
Describe 'Test Should -Throw' {
    It '1' {
        { throw "This is a test" } | Should -Throw 'This is a test' # Test will pass
    }
    it '2' {
        {throw 'this is a test'} | Should -Throw 'test' # Test will pass
    }
    it '3' {
        {throw 'this is a test'} | Should -Throw "*is*" # Test will not pass
    }
    it '4' {
        {throw 'this is a test'} | Should -Throw "*is*test" # Test will not pass
    }
}
```

It would be great if all these tests could be evaluated as True.

## 4.Current Behavior

Test 3 and 4 fail.

**Comment by alx9r** [comment]
It looks like [this is the line](https://github.com/pester/Pester/blob/b8fa450b30eacce1318c9a0e81f5617f6babc471/Functions/Assertions/PesterThrow.ps1#L126) that currently does the comparison matching.  It uses `[string].IndexOf()` to do the pattern matching. That seems like a curious way of matching strings.  It looks like [it was changed from `.Contains()`](https://github.com/pester/Pester/commit/2cd2eaf8b0bdc1ead464d27acf3615ce7ca2cac9#diff-22787d4946053ce52939c98fe2a9b19cR56) because "Contains returns $True for $null input".  [Previous to that it used -cnotlike](https://github.com/pester/Pester/commit/c10cb962c38d66610a8cab61a4dda4c8e06b9d74#diff-22787d4946053ce52939c98fe2a9b19cL15).

It seems like matching on wildcards like `*is*test` could be achieved by using `-like`.  If that change were made for all calls to `Should -throw` I expect it would break some existing tests that contained `*`, `?`, or `[` in their existing messages, though.  Those characters are the PowerShell wildcard characters per 3.15 of [the PowerShell spec](https://www.microsoft.com/en-ca/download/details.aspx?id=36389).  It seems like implementing this safely would involve user-selection of the matching mechanism.

**Comment by dlwyatt** [comment]
In v4, we can just add some switches to control the comparison behavior, similar to how the switch statement works (-Regex and -Wildcard options).  Yay for dynamic parameter sets.  :)

**Comment by nohwnd** [comment]
Imho reverting back to `-cnotlike` is the most sensible thing to do (I actually thought it still works like that. :) ). For anything more advanced you can pipe the exception object via `-PassThru` and do whatever you like.

**Comment by DarkLite1** [comment]
Thanks for the suggestion @nohwnd , for now I fixed it like this:

```powershell
{throw 'this is a test'} | 
Should -Throw -PassThru | Select-Object -ExpandProperty Exception | 
Should -BeLike "*is*test*"
```

**Comment by nohwnd** [comment]
Marked this as Help wanted. Implementers please revert back to use `-cnotlike` to accomodate the requeste and avoid breaking change.

**Comment by fflaten** [comment]
@nohwnd Shouldn't this be considered a breaking change if it's done without a new `-Wildcard` switch? Or did you agree on the switch? :)

 `Contains()` and `-cnotlike` do not behave the same. `Contains()` would accept `test` as a pattern, while `-*like` requires wildcards. The change was done 6 years ago, so there might be a lot of users who expect the current behaviour.

```
'this is a test'.Contains("*is*test")
False
PS C:\Users\frode> -not ('this is a test' -cnotlike "*is*test")
True
PS C:\Users\frode> 'this is a test'.Contains("a test")
True
PS C:\Users\frode> -not ('this is a test' -cnotlike "a test")
False
```

**Comment by nohwnd** [comment]
@fflaten kk, one more thing to do in Pester v5.

**Comment by nohwnd** [comment]
fixed by #1287
